vmdb/app/models/miq_provision_task_configured_system_foreman/state_machine.rb

```diff
@@ -0,0 +1,73 @@
+require 'manageiq_foreman'
+module MiqProvisionTaskConfiguredSystemForeman::StateMachine
+  def run_provision
+    set_source_from_options
+    signal :prepare_provision
+  end
+
+  def prepare_provision
+    prepare_provider_options
+    merge_provider_options_from_automate
+    signal :start_configuration_task
+  end
+
+  def start_configuration_task
+    update_and_notify_parent(:message => "Starting configuration of #{source.name}")
+    log_provider_options
+    signal :system_power_off_in_foreman
+  end
+
+  def system_power_off_in_foreman
+    update_and_notify_parent(:message => "Powering off #{source.name}")
+    source.provider_object.stop
+    signal :poll_system_powered_off_in_foreman
+  end
+
+  def poll_system_powered_off_in_foreman
+    update_and_notify_parent(:message => "Waiting for power off of #{source.name}")
+    if source.provider_object.powered_off?
+      signal :update_configuration
+    else
+      requeue_phase
+    end
+  end
+
+  def update_configuration
+    update_and_notify_parent(:message => "Updating configuration of #{source.name}")
+    source.provider_object.update(phase_context[:provider_options])
+    signal :os_build
+  end
+
+  def os_build
+    update_and_notify_parent(:message => "Building OS on #{source.name}")
+    source.provider_object.start
+    signal :poll_os_built
+  end
+
+  def poll_os_built
+    update_and_notify_parent(:message => "Waiting for OS build on #{source.name}")
+    if source.provider_object.building?
+      requeue_phase
+    else
+      EmsRefresh.queue_refresh(source)
+      signal :mark_as_completed
+    end
+  end
+
+  def mark_as_completed
+    begin
+      MiqEvent.raise_evm_event(source, 'vm_provisioned', :configured_system => source)
```

Comment:
We will want to raise a different event here like `configured_system_provisioned` which will need to be defined in db/fixtures/miq_events.csv.  Feels like it needs a new event category as well which comes from db/fixtures/miq_event_sets.csv.